ml/src/test/scala/ai/h2o/sparkling/ml/models/H2OMOJOPipelineModelTestSuite.scala

```diff
@@ -294,4 +297,51 @@ class H2OMOJOPipelineModelTestSuite extends FunSuite with SparkTestContext {
     val settings = H2OMOJOSettings()
     testTransformAndTransformSchemaAreAligned(settings)
   }
+
+  test("Test distributed pipeline") {
+    // Test mojo
+    // Notes:
+    //   - Need to disable stage codegen: 'spark.sql.codegen.wholeStage' = false
+    //   - Also need schema from MOJO to assembly proper inputs for algos
+    //   - there is no way how to get output columns from MOJO at this point
+    //   - TODO: the MOJO should provide training schema, also scoring schema
+    val mojoTransf = H2OMOJOPipelineModel.createFromMojo(
+      "/tmp/mojo-pipeline/pipeline.mojo",
+      H2OMOJOSettings(removeModel = true, expandNamedMojoOutputColumns = true))
+
+    val inputScoringSchema = StructType(mojoTransf.getFeaturesCols().map(name => StructField(name, DoubleType, nullable = true)))
+    val inputTrainingSchema =
+      StructType(inputScoringSchema.fields ++ Array(StructField("default payment next month", IntegerType, nullable = true)))
+
+    val trainingDataset = spark.read
+      .schema(StructType(Array(StructField("ID", IntegerType, nullable = false)) ++ inputTrainingSchema.fields))
```

Comment:
@mmalohlava any reason why we don't let Spark to infer schema? 

Is it becuase of the optimization -> settting right types from the beggining?